Source/WebKit/Platform/cocoa/ExtensionCapabilityGrant.mm

```diff
@@ -35,14 +35,21 @@
 
 namespace WebKit {
 
-static void platformInvalidate(id<BEProcessCapabilityGrant> platformGrant)
+static void platformInvalidate(PlatformGrant platformGrant)
 {
-    if (![platformGrant isValid])
-        return;
-
 #if USE(EXTENSIONKIT)
-    if (![platformGrant invalidate])
-        RELEASE_LOG_ERROR(ProcessCapabilities, "Invalidating grant %{public}@ failed", platformGrant);
+    WTF::switchOn(platformGrant, [&] (auto& grant) {
+        if (![grant isValid])
+            return;
```

Comment:
This returns from the lambda function, not from the platformInvalidate() function. As a result, your switchOn() here is a no-op and we no longer early return when the grant isn't valid.